### DISCUSSION TITLE
How to get well formed JSON from request body?

### DISCUSSION THREAD

**Discussion by aplatypus** [testingisdocumenting/webtau#1156] [Q&A]
* I have sent a request to a REST servlet and it returns a JSON array
  * I need to get the JSON as a String for further processing.
* Originally I used the 
   `DataNode  result =  body.get("caaList");`
   `return result.toString();`
* Apparently this is an exception according tothe `Jackson` library:
  ```
   "Unexpected character ('c' (code 99)): was expecting double-quote to start field name
    at [Source: (String)"[{caaNumber: 22248002, toaNumber: }]"; line: 1, column: 4]"
  ```
  * These are the expected values.
  * They are not in legitimate JSON form.
* I've done some rooting about in the documentation, I have seen many useful xpath like calls to get specific values or even lists.  
  * I have not seen anything to get me the (raw) JSON from the response
  * For the example above the requirement is: `[ {"caaNumber" : "1234", "toaNumber" : "567"}, ...  ]`
  * If that's not available the whole JSON content for the body would be usable:
    `"caaList" : [ {"caaNumber" : "1234", "toaNumber" : "567"}, ...  ]`, I can work with that.
* I expect I don't know my way around the documentaton and examples well enough
* While now that I know the `toString()` method just gives me a map-like structure I could make a StringBuilder and remake-up the data to JSON.  However, that almost defeats the purppse of the end-to-end REST testing.
  *  That JSON format is part of the protocol we are testing.
  *  I feel like the answer is there staring at me... Like `body.get()` or `body.getRaw()`

I should stop speculating and leave the question as is.  How to get legitimate JSON content from a `DataNode`?

**Comment by MykolaGolubyev** [comment]
This is a good question, thank you. I will have to update the documentation to include the answer. 
Also any chance we can follow-up your earlier suggestion and move questions like this to Stack Overflow? Will help promote the tool as well and I will subscribe to the tag on stack overflow

**Reply by MykolaGolubyev** [reply]
also could you please elaborate on "That JSON format is part of the protocol we are testing."? Are you validating that JSON matches a certain schema? or that the json is well formed?

**Reply by aplatypus** [reply]
Righty-o ... Stackoverflow it is.  It might get some community input too on how to get what I want smarter too.  Seem that stackoverflow is the go.

 * [Where should we ask software testing questions?](https://meta.stackexchange.com/questions/55782/where-should-we-ask-software-testing-questions)
   * Questions can be tagged, as appropriate: [WebTau](https://stackoverflow.com/questions/tagged/webtau), [Unit-Testing](https://stackoverflow.com/questions/tagged/unit-testing) and [testing](https://stackoverflow.com/questions/tagged?tagnames=testing&sort=newest&pagesize=50)

My well formed JSON comment is a simple result of the tooling.  When a string isn't correct JSON syntax I get a free Exception.  Sometimes that is enough I suppose.  But if the body is correct JSON syntax, then the error is coming from the test ahrness, that's not desirable.

There's no link to the place to look in the documentation, so I'll repost this query on Stackoverflow with a [`webtau`] tag to get the ball rolling.  The repost is here:
* [How to retrieve JSON from request body using WebTau?](https://stackoverflow.com/questions/72567932/how-to-retrieve-json-from-request-body-using-webtau)
